File: leetcode/633. Sum of Square Numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        for a in range(0, int(c ** 0.5) + 1):
            b = (c - a ** 2) ** 0.5
            if int(b) == b:
                return True
            if b <= a:
                return False
        return False
        

File: leetcode/633. Sum of Square Numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        for(int a = 0; a < int(sqrt(c)) + 1; ++a){
            double b = sqrt(c - pow(a, 2));
            if(b == int(b)){
                return true;
            }
            if(b <= a){
                return false;
            }
        }
        return false;
    }
};
